src/internal/packager/helm/post-render.go

```diff
@@ -29,21 +32,34 @@ type renderer struct {
 	*Helm
 	connectStrings types.ConnectStrings
 	namespaces     map[string]*corev1.Namespace
+	ctx            context.Context
 }
 
 func (h *Helm) newRenderer() (*renderer, error) {
 	message.Debugf("helm.NewRenderer()")
 
-	namespaces := make(map[string]*corev1.Namespace)
-	if h.cluster != nil {
-		namespaces[h.chart.Namespace] = h.cluster.NewZarfManagedNamespace(h.chart.Namespace)
+	rend := &renderer{
+		Helm:           h,
+		connectStrings: types.ConnectStrings{},
+		namespaces:     map[string]*corev1.Namespace{},
+		ctx:            context.Background(),
```

Comment:
The `context` package docs recommend to not store contexts inside a struct type and to pass it as the first argument to the function:

> Programs that use Contexts should follow these rules to keep interfaces consistent across packages and enable static analysis tools to check context propagation:

> Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:

```go
func DoSomething(ctx context.Context, arg Arg) error {
	// ... use ctx ...
}
```
https://pkg.go.dev/context#pkg-overview

Reply:
Thanks for linking those docs, I was trying to pass the context through to the `Run()` function which satisfies a helm interface so it can't take context, but yeah we're not doing anything with the context anyway and it's better to follow go convention. I'm just going to continue creating the context in those spots